packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -122,24 +173,50 @@ export const Stackbox = () => {
   const [showInsufficentBalanceError, setShowInsufficentBalanceError] =
     useState(false);
 
-  const isStrategySelected = Boolean(selectedStrategy);
+  const { data: balance, isRefetching: isRefetchingBalance } = useBalance({
+    address: Boolean(fromToken) ? address : undefined,
+    token: fromToken?.address as `0x${string}`,
+    chainId: chain?.id,
+  });
 
   useEffect(() => {
-    if (!selectedStrategy)
-      setEndDateTime(new Date(endDateByFrequency[frequency]));
+    if (isTokenListLoading) return;
+
+    const searchParams = new URLSearchParams(window.location.search);
+
+    const fromTokenParam = searchParams.get("fromToken");
+    if (
+      fromTokenParam &&
+      fromToken.address.toLowerCase() != fromTokenParam.toLowerCase()
+    ) {
+      setFromToken(getTokenFromList(fromTokenParam));
+    }
+
+    const toTokenParam = searchParams.get("toToken");
+    if (
+      toTokenParam &&
+      toToken.address.toLowerCase() != toTokenParam.toLowerCase()
+    ) {
+      setToToken(getTokenFromList(toTokenParam));
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [frequency]);
+  }, [isTokenListLoading]);
 
-  // set default tokens
   useEffect(() => {
-    const chainId = chain?.unsupported
-      ? ChainId.GNOSIS
-      : chain?.id ?? ChainId.GNOSIS;
+    if (chain) setChainId(chain.id);
+  }, [chain, chain?.id, setChainId]);
 
-    const { from, to } = DEFAULT_TOKENS_BY_CHAIN[chainId];
-    setFromToken(from);
-    setToToken(to);
-  }, [chain]);
+  useEffect(() => {
+    if (
+      chainId &&
+      switchNetwork &&
+      chainId !== chain?.id &&
+      Boolean(ChainId[chainId])
+    ) {
+      switchNetwork(chainId);
+    }
```

Comment:
**MINOR**, non-blocking:
No need to check for `chainId` if we're going to use it in a logic comparison, as it would default into a falsy value if it's `undefined`, and then the other checks would handle the setter execution
```suggestion
if (chainId !== chain?.id && Boolean(ChainId[chainId]) && switchNetwork) {
      switchNetwork(chainId);
    }
```

Reply:
Typescript doesn't allow this because `chainId` type is `number | null` and we can't check an enum with null.

Reply:
I'm starting to guess we have diff Typescript and Typechecks running.
When I checked this locally it worked w/o warnings even in console